src/test/bridges/liquity/StakingBridgeUnit.t.sol

```diff
@@ -37,42 +39,49 @@ contract StakingBridgeTest is TestUtil {
 
     function testIncorrectInput() public {
         // Call convert with incorrect input
-        vm.prank(address(rollupProcessor));
         vm.expectRevert(ErrorLib.InvalidInput.selector);
-        bridge.convert(
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            0,
-            0,
-            0,
-            address(0)
-        );
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
     }
 
     function testFullDepositWithdrawalFlow() public {
         // I will deposit and withdraw 1 million LQTY
-        uint256 lqtyAmount = 1e24;
-        _deposit(lqtyAmount);
+        uint256 inputValue = 1e24;
+        _deposit(inputValue);
 
-        // Withdraw LQTY from the staking contract through the bridge
-        rollupProcessor.convert(
+        AztecTypes.AztecAsset memory inputAssetA = AztecTypes.AztecAsset(
+            2,
             address(bridge),
-            AztecTypes.AztecAsset(2, address(bridge), AztecTypes.AztecAssetType.ERC20),
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            AztecTypes.AztecAsset(1, tokens["LQTY"].addr, AztecTypes.AztecAssetType.ERC20),
-            AztecTypes.AztecAsset(0, address(0), AztecTypes.AztecAssetType.NOT_USED),
-            lqtyAmount,
+            AztecTypes.AztecAssetType.ERC20
+        );
+        AztecTypes.AztecAsset memory outputAssetA = AztecTypes.AztecAsset(
             1,
-            0
+            tokens["LQTY"].addr,
+            AztecTypes.AztecAssetType.ERC20
+        );
+
+        // Transfer SB back to the bridge
+        IERC20(inputAssetA.erc20Address).transfer(address(bridge), inputValue);
+
+        // Withdraw LQTY from the staking contract through the bridge
+        (uint256 outputValueA, , ) = bridge.convert(
+            inputAssetA,
+            emptyAsset,
+            outputAssetA,
+            emptyAsset,
+            inputValue,
+            1,
+            0,
+            address(0)
         );
 
         // Check the total supply of SPB token is 0
         assertEq(bridge.totalSupply(), 0);
 
-        // Check the LQTY balance of rollupProcessor is equal to the initial LQTY deposit
-        assertEq(tokens["LQTY"].erc.balanceOf(address(rollupProcessor)), lqtyAmount);
+        // Transfer the funds back from the bridge to the rollup processor
+        IERC20(outputAssetA.erc20Address).transferFrom(address(bridge), rollupProcessor, outputValueA);
+
+        // Check the LQTY balance of rollupProcessor is greater than or equal to the initial LQTY deposit
+        assertGe(outputValueA, inputValue);
     }
 
     function testMultipleDepositsWithdrawals() public {
```

Comment:
It doesn't make sense here because staking accrual comes from borrowing and redemption fees, not token emission. For this reason I am testing the functionality by simply minting the rewards to the bridge.
<img width="749" alt="image" src="https://user-images.githubusercontent.com/13470840/178430562-9483e282-6e88-44f3-b515-f97c16bc1bf6.png">